src/unity/python/turicreate/toolkits/drawing_recognition/_sframe_loader.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2019 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+from __future__ import print_function as _
+from __future__ import division as _
+from __future__ import absolute_import as _
+import numpy as _np
+import turicreate as _tc
+import mxnet as _mx
+from six.moves.queue import Queue as _Queue
+from threading import Thread as _Thread
+from turicreate.toolkits._main import ToolkitError as _ToolkitError
+
+_TMP_COL_RANDOM_ORDER = '_random_order'
+
+# Encapsulates an SFrame, iterating over each row and returning an
+# (image, label, index) tuple.
+class _SFrameDataSource:
+    def __init__(self, sframe, feature_column, target_column,
+                 load_labels=True, shuffle=True, samples=None):
+        self.target_column = target_column
+        self.feature_column = feature_column
+        self.load_labels = load_labels
+        self.shuffle = shuffle
+        self.num_samples = samples
+        self.cur_sample = 0
+
+        # Make shallow copy, so that temporary columns do not change input
+        self.sframe = sframe.copy()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self._next()
+
+    def next(self):
+        return self._next()
+
+    def _next(self):
+        if self.cur_sample == self.num_samples:
+            raise StopIteration
+
+        # If we're about to begin a new epoch, shuffle the SFrame if requested.
+        row_index = self.cur_sample % len(self.sframe)
+        if row_index == 0 and self.cur_sample > 0 and self.shuffle:
+            self.sframe[_TMP_COL_RANDOM_ORDER] = _np.random.uniform(size=len(self.sframe))
+            self.sframe = self.sframe.sort(_TMP_COL_RANDOM_ORDER)
+        self.cur_sample += 1
+
+        # Copy the image data for this row into a NumPy array.
+        row = self.sframe[row_index]
+
+        drawing_feature = row[self.feature_column]
+        is_stroke_input = (type(drawing_feature) != _tc.Image)
+        if is_stroke_input:
+            pixels_01 = drawing_feature
```

Comment:
Didn't you just test that `drawing_feature` is NOT an image here? How are you interpreting it as pixels? Should there be a TODO here to implement rasterization?

Reply:
This is a little confusing. 

In this PR, _drawing_recognition_prepare_data in C++ returns a flex_list of pixel values, which can directly be passed in to the neural network which is why the sframe_loader doesn't do anything.

As of #1449,  _drawing_recognition_prepare_data in C++ returns a flex_image, which sframe_loader would need to extract pixel_data from.